data: add support of anydata into dict_to_dnode

steweg opened the pull request:
This patch adds support of anydata within dict_to_dnode. To do that new options were introduced into dict_to_dnode() API

samuel-gauthier commented:
@steweg, what do you want to do with this draft? Shall we close it or do you have questions?

steweg commented:
I will need to rebase it and then I will mark it for review

steweg commented:
This one needs first #107 to be merged

steweg commented:
I have rebased based on current #107 for the time being (first 3 commits are from #107). It can at least be reviewed and rebased based on master later on.

steweg commented:
@samuel-gauthier I have rebased the patch based on current master. This PR is ready for review
